src/ABC_Datalog/main.pl

```diff
@@ -0,0 +1,447 @@
+/*
+Date: 07 Jan 2019
+Macintosh HD⁩/⁨Users⁩/lixue⁩/GoogleDrive⁩/01PHD⁩/01program⁩/eclipse-workspace⁩/ABC_Clean⁩/src⁩
+*/
+
+:- use_module(library(lists)).
+:-[preprocess, equalities, repairPlanGen, repairApply, vitality, fileOps].
+    % clear all assertions. So main has to be compiling before the input theory file.
+:-    maplist(retractall, [trueSet(_), falseSet(_), heuristics(_), protect(_), spec(_)]).
+
+/********************************************************************************************************************** Global Variable and their values.
+debugMode:    0 -- no write_term_c information.
+            1 -- write_term_c informaiton.
+spec(pft(:        the true set of the preferred structure in the internal format.
+spec(pff:        the false set of the preferrd structure in the internal format.
+costLimit:    the maximum length of the repair plans.
+Signature:  [predicateInfoList, ConstantLists], where
+            predicateInfoList = [(predsymbol, Arity Info, Arguments Domain), ...]
+            Arity Info:  [(arity1, source1), (arity2, source2)], e.g., mum, [(3, theory), (2, prefStruc)].
+            Arguments Domain: [[c11,c12..], [c21, c22,..]...], where [c11,c12..] is a list of constants occur as the first argument of that predicate in all theorems of P.
+proofStatus:  0 -- default value.
+            1 -- will get an axiom for resolving the goal, but there is no more axiom in the input theory.
+            2 -- a positive literal is derived.
+**********************************************************************************************************************/
+
+:-dynamic pause/0.
+%:-spy(pause).
+pause.
+
+% if there is any rules in the ps
+abc:-
```

Comment:
This should now be called main inside the file of abc.pl. But I will fix this issue for you.